tests/unit/backend/wmg/api/test_expression_rollup.py

```diff
@@ -54,14 +56,326 @@ def test__cell_types_in_same_lineage_are_colinear(self):
             a, b = cell_types
             assert are_cell_types_colinear(a, b) == expected
 
-    def test__missing_combinations_added_to_expression_dataframe_before_rollup(self):
+
+def _create_cell_counts_df_helper(cell_counts_rows: list[list], columns: list[str], index_cols: list[str]) -> DataFrame:
+    cell_counts_df = pd.DataFrame(cell_counts_rows, columns=columns)
+    cell_counts_df = cell_counts_df.set_index(index_cols)
```

Comment:
Didn't know you could do this - so if you pass a dataframe a list of column names in `set_index`, it creates a multi-index automatically? What happens if there are duplicate combinations in the passed index columns?

Reply:
@atarashansky - It is not clear but _I think_ it might throw exception if you set `verify_integrity` argument to `True`: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.set_index.html

For this case, I wanted to create a dataframe containing the rows in `cell_counts_rows` with the multi-index. The assumption (because this is a test) is that rows can be properly multi-indexed in the way specified.

To make it clear, I can set `verify_integrity=True` to make the expectations about the rows clear.